### PR TITLE
verify_apo robot re-written

### LINKED ISSUE #12: (IR) re-implement and re-name verify-agreement robot
We probably want this called something like `verify-policy`.   Check what Lynn put in the workflow DF (and change it if you have a better name, after discussing) -- see #4

current code resides in
- sdr-preservation-core (called verify-agreement)
- ???

any code in archive-utils or sdr-replication should either go in preservation_robots or moab-versioning or ... ; 

intend to deprecate:
- sdr-preservation-core
- archive-utils
- sdr-replication

#### from sdr-preservation-core, per sul-dlss/preservation2017#45:

>**(5) verify-agreement** [code](https://github.com/sul-dlss/sdr-preservation-core/blob/master/lib/sdr_ingest/verify_agreement.rb)
>- These seem to be fairly self-explanatory and need no functional change. It is actually the admin policy that is being verified, not the agreement, so a name change could clarify that.
>- Recommend: no functional change; optional name change and code-references updated from agreement to APO.

### PR DESCRIPTION
**Pull request by ndushay**
(at least 2 reviewers - but it's not overwhelming)

This is a replacement for the old robot code here:  https://github.com/sul-dlss/sdr-preservation-core/blob/master/lib/sdr_ingest/verify_agreement.rb

What to think about:

- does the code have equivalent functionality as the original?
    -  except, by choice, this version is not trying to cache information;  experimentation is planned with another branch to see if this should be added in another PR
- does the code make sense to you?
- are the tests checking the right stuff?
- do the tests make sense to you?
- are the tests at least as thorough as the original tests here: https://github.com/sul-dlss/sdr-preservation-core/blob/master/spec/sdr_ingest/verify_agreement_spec.rb  (which seem to be a whole lot of mocking and not a lot of actual testing)

Closes #12

### CODE REVIEW

**Comment by atz** [comment]
I appreciate all the conditionals getting flattened, especially the `raise` guard clauses.

File: robots/preservation/verify_apo.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    def apo_druid
+      rel_md_ng_xml = Nokogiri::XML(File.open(relationship_md_pathname.to_s), &:strict)
+      nodeset = rel_md_ng_xml.xpath("//hydra:isGovernedBy", 'hydra' => 'http://projecthydra.org/ns/relations#')
+      raise(ItemError, "Unable to find isGovernedBy node of relationshipMetadata") if nodeset.empty?
+
+      apo_attr = nodeset.first.attribute_with_ns('resource', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#')
+      raise(ItemError, "Unable to find 'resource' attribute for <isGovernedBy> in relationshipMetadata") unless apo_attr
+      apo_attr.text.split('/')[-1]
+    rescue Nokogiri::XML::SyntaxError => e
+      raise(ItemError, "Unable to parse #{relationship_md_pathname}: #{e.message}")
```
Comment (atz) [inline robots/preservation/verify_apo.rb]: 👍 Including the original exception's message!!

File: spec/preservation/verify_apo_spec.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
+describe Preservation::VerifyApo do
+  let(:deposit_dir_pathname) { Pathname(File.join(File.dirname(__FILE__), '..', 'fixtures', 'deposit')) }
+  let(:verify_apo) { described_class.new }
```
Comment (atz) [inline spec/preservation/verify_apo_spec.rb]: Probably could declare this your `subject(:verify_apo)`, if you like.

**Comment by ndushay** [comment]
@atz pointed out that I had missed part of his suggestions;  I just corrected that.

**Pull request merged**